pre-commit-hooks/generate-mockery.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+# vi: ts=2:sw=2:et
+# vim: ft=bash
+
+set -euo pipefail
+
+if ! command -v mockery &> /dev/null
+then
+  echo "mockery is not installed or is not in the current PATH" >&2
+  echo "Find out more at: https://github.com/vektra/mockery" >&2
+  exit 2
+fi
+
+if ! git diff --exit-code mocks &> /dev/null
```

Comment:
I'm a bit worried by this. The folder is not necessarily in mocks folder at the root folder.

Mockery can have a config file.

Are you sure you have to check the status of modified files?

I think I remember pre-commit does itself.

Reply:
The part I'm worried about is the workflow of:
I have changes staged for commit, and some unstaged because I like small specific commits.
Some of my unstaged commits may also need their mockery updated.
The mockery does update my staged files and unstaged files.
How does the pre-commit deal with this situation.

I'm not saying that what I have here does that. But it's a thought that I had.